app/javascript/components/app.jsx

```diff
@@ -75,7 +76,16 @@ const client = new ApolloClient({
 const App = () => (
   <ApolloProvider client={client}>
     <GlobalStyle />
-    <WidgetController client={client} />
+    <Router>
+      <Switch>
+        <Route
+          path="/:city_name"
+          component={routeProps => (
+            <WidgetController client={client} location={routeProps.location} />
```

Comment:
`routeProps` should contain `match` generated from the pattern(s) in `path` above:
https://reactrouter.com/web/api/match 

This would skip the querystring logic from leaking into our component down below with `location.pathname.substring(1)`

Reply:
I agree. I would suggest a hooks solution so we don't need to put it in props at all. Lets also rename this since `location` is getting to overloaded.

```jsx
import { useRouteMatch } from "react-router-dom";

function useCurrentCityName(){
  const { params: {city_name} } =  useRouteMatch("/:city_name");
  return city_name;
}
```

Reply:
I see what you are saying, will make those changes

Reply:
I think route props provides transparency over hooks.